win32/src/win32evtlog.i

```diff
@@ -511,6 +512,114 @@ static PyObject *PyEvtOpenChannelEnum(PyObject *self, PyObject *args, PyObject *
 }
 PyCFunction pfnPyEvtOpenChannelEnum = (PyCFunction) PyEvtOpenChannelEnum;
 
+
+// Helper function to convert a list of strings double zero terminated
+// into a Python list of strings.
+// e.g. hello\0world\0\0 -> [ "hello", "world" ]
+static PyObject* PyList_FromDoubleTerminatedWSTR(LPWSTR strings)
+{
+	PyObject* ret = PyList_New(0);
+	if (ret == NULL) {
+		return NULL;
+	}
+
+	WCHAR* cur = strings;
+	while (*cur) {
+		PyObject* keyword = PyWinObject_FromWCHAR(cur);
+		PyList_Append(ret, keyword);
+		Py_XDECREF(keyword);
+		cur += wcslen(cur) + 1;
+	}
+
+	return ret;
+}
+
+// Used internally to format event messages
+// Caller must free the returned value
+static PyObject *FormatMessageInternal(EVT_HANDLE metadata, EVT_HANDLE event, DWORD flags, DWORD resourceId)
+{
+	LPWSTR buf = NULL;
+	PyObject *ret = NULL;
+	DWORD allocated_size = 0;
+	DWORD returned_size = 0;
+	DWORD status = 0;
+	DWORD err = 0;
+
+	BOOL bsuccess = 0;
+	Py_BEGIN_ALLOW_THREADS
+	// Get the size of the buffer
+	bsuccess = EvtFormatMessage(metadata, event, resourceId, 0, NULL, flags, allocated_size, buf, &returned_size);
+	Py_END_ALLOW_THREADS
+
+	err = GetLastError();
+
+	// The above call should always return ERROR_INSUFFICIENT_BUFFER
+	if (!bsuccess && err != ERROR_INSUFFICIENT_BUFFER) {
+		PyWin_SetAPIError("EvtFormatMessage");
```

Comment:
you need to `return NULL` here (or, IIRC, just `return PyWin_SetAPIError("EvtFormatMessage");`)